commercetools/resource_channel_test.go

```diff
@@ -0,0 +1,345 @@
+					),
+					resource.TestCheckResourceAttr(
+						"commercetools_channel.standard", "custom.0.field.1.key", "meal",
+					),
+					resource.TestCheckResourceAttr(
+						"commercetools_channel.standard", "custom.0.field.1.value", "{\"de-DE\":\"Mittag\",\"en-GB\":\"lunch\"}",
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccChannelConfig() string {
+	return `
+
+resource "commercetools_type" "channel_test" {
+	key = "channel-test"
+
+	resource_type_ids = ["channel"]
+
+	field {
+		name = "carrier"
+		label = {
+			en = "Skype name"
+			nl = "Skype naam"
+		}
+		type {
+			name = "String"
+		}
+	}
+
+	field {
+		name = "meal"
+		label = {
+			en = "Skype name"
+			nl = "Skype naam"
+		}
+		type {
+			name = "LocalizedString"
+		}
+	}
+
+	name = {
+		en = "Contact info"
+		nl = "Contact informatie"
+	}
+	description = {
+		en = "All things related communication"
+		nl = "Alle communicatie-gerelateerde zaken"
+	}
+}
+
+resource "commercetools_channel" "standard" {
+	 depends_on = [
+		commercetools_type.channel_test,
+	  ]
+	roles = ["Primary"]
+	key  = "standard-key"
+ 	custom {
+		type_key = "channel-test"
+		field {
+		  key = "carrier"
+		  value = jsonencode("example")
+		}
+
+		field {
+		  key = "meal"
+		  value = jsonencode({
+			"en-GB": "lunch",
+		  })
+		}
+	}
+}
+`
+}
+
+func testAccChannelConfigWithoutCustom() string {
+	return `
+
+resource "commercetools_type" "channel_test" {
+	key = "channel-test"
+
+	resource_type_ids = ["channel"]
+
+	field {
+		name = "carrier"
+		label = {
+			en = "Skype name"
+			nl = "Skype naam"
+		}
+		type {
+			name = "String"
+		}
+	}
+
+	field {
+		name = "meal"
+		label = {
+			en = "Skype name"
+			nl = "Skype naam"
+		}
+		type {
+			name = "LocalizedString"
+		}
+	}
+
+	name = {
+		en = "Contact info"
+		nl = "Contact informatie"
+	}
+	description = {
+		en = "All things related communication"
+		nl = "Alle communicatie-gerelateerde zaken"
+	}
+}
+
+resource "commercetools_channel" "standard" {
+	 depends_on = [
+		commercetools_type.channel_test,
+	  ]
+	roles = ["Primary"]
+	key  = "standard-key"
+}
+`
+}
+
+func testAccChannelUpdateConfig() string {
+	return `
+
+resource "commercetools_type" "channel_test" {
+	key = "channel-test"
+
+	resource_type_ids = ["channel"]
+
+	field {
+		name = "carrier"
+		label = {
+			en = "Skype name"
+			nl = "Skype naam"
+		}
+		type {
+			name = "String"
+		}
+	}
+
+	field {
+		name = "meal"
+		label = {
+			en = "Skype name"
+			nl = "Skype naam"
+		}
+		type {
+			name = "LocalizedString"
+		}
+	}
+
+	name = {
+		en = "Contact info"
+		nl = "Contact informatie"
+	}
+	description = {
+		en = "All things related communication"
+		nl = "Alle communicatie-gerelateerde zaken"
+	}
+}
+
+resource "commercetools_channel" "standard" {
+	 depends_on = [
+		commercetools_type.channel_test,
+	  ]
+	roles = ["Primary"]
+	key  = "standard-key"
+ 	custom {
+		type_key = "channel-test"
+
+		field {
+		  key = "carrier"
```

Comment:
should this perhaps be named `name = "carrier"` to be consistent?

Reply:
can change it also to name, I don't have any strong feelings about it. Will change it tomorrow